src/ir/from_ast/astconv.rs

```diff
@@ -355,8 +374,17 @@ impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
         let param = self.comp.get_mut(p.live.idx);
         param.owner = ir::ParamOwner::bundle(idx);
 
+        // If this is a signature port, try adding it to the component's external interface
+        if is_sig_port {
```

Comment:
Inline the definition of the variable and the use since it isn't that complicated an expression

Reply:
I can't because it has to borrow comp as immutable, which is also needed as mutable further into the statement.